cmdstanpy/utils.py

```diff
@@ -216,6 +216,56 @@ def rdump(path: str, data: Dict) -> None:
             fd.write('\n')
 
 
+def rload(fname: str) -> dict:
+    """Parse data and parameter variable values from an R dump format file.
+       This parser only supports the subset of R dump data as described
+       in the "Dump Data Format" section of the CmdStan manual, i.e.,
+       scalar, vector, matrix, and array data types.
+    """
+    data_dict = {}
+    with open(fname, 'r') as fp:
+        lines = fp.readlines()
+
+    # Variable data may span multiple lines, parse accordingly
+    idx = 0
+    while idx < len(lines) and '<-' not in lines[idx]:
+        idx += 1
+    if idx == len(lines):
+        return None
+    start_idx = idx
+    idx += 1
+    while True:
+        while idx < len(lines) and '<-' not in lines[idx]:
+            idx += 1
+        next_var = idx
+        var_data = ''.join(lines[start_idx:next_var]).replace('\n', '')
+        lhs, rhs = [_.strip() for _ in var_data.split('<-')]
+        lhs = lhs.replace('"','')
+        rhs = rhs.replace('L','')
+        if rhs.startswith('structure'):
+            *_, vals, dim = rhs.replace('(', ' ').replace(')', ' ').split('c')
+            vals = [float(v) for v in vals.split(',')[:-1]]
+            dim = [int(v) for v in dim.split(',')]
+            val = np.array(vals).reshape(dim[::-1]).T
```

Comment:
what is going on here?

Would this equal

    np.array(vals).reshape(dim[::-1]).T 
    --> np.array(vals, order='F').reshape(dim, order='F')
    or
    --> np.array(vals).reshape(dim, order='F')
    or
    --> np.reshape(vals, dim, order='F')

Reply:
agreed, that's not very clear.  I added a subroutine that processes the Rdump multi-dim structure - used an ugly regex with named groups for the essential bits - array values, array dimensions.   I ran the  Rdump file in R to see what R does, then added unit tests that verify that the array values end up in the right place.   hope this helps.